test/e2e/case12_ordering_test.go

```diff
@@ -0,0 +1,385 @@
+				true,
+				defaultTimeoutSeconds)
+
+			return getCompliant(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+
+		By("Generating a non compliant event on the dep")
+		managedPlc = utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Warning",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"NonCompliant; there is violation")
+
+		By("Checking if dependency status is noncompliant")
+		Eventually(checkCompliance(case12DepName), defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+
+		By("Checking if policy status is pending")
+		Eventually(checkCompliance(case12PolicyName), defaultTimeoutSeconds, 1).Should(Equal("Pending"))
+	})
+	It("Should process extra dependencies properly", func() {
+		By("Creating an extra dep on hub cluster in ns:" + clusterNamespaceOnHub)
+		_, err := kubectlHub("apply", "-f", case12DepBYaml, "-n", clusterNamespaceOnHub)
+		Expect(err).To(BeNil())
+		hubPlc := utils.GetWithTimeout(
+			clientHubDynamic,
+			gvrPolicy,
+			case12DepBName,
+			clusterNamespaceOnHub,
+			true,
+			defaultTimeoutSeconds)
+		Expect(hubPlc).NotTo(BeNil())
+
+		By("Generating a non compliant event on the dep b")
+		managedPlc := utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepBName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Warning",
+			"policy: managed/namespace-foo-setup-configpolicy-b",
+			"NonCompliant; there is violation")
+
+		By("Checking if dependency status is noncompliant")
+		Eventually(checkCompliance(case12DepBName), defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+
+		By("Generating an compliant event on the dependency")
+		managedPlc = utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Normal",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"Compliant; No violation detected")
+		By("Checking if dependency status is compliant")
+		Eventually(func() interface{} {
+			managedPlc = utils.GetWithTimeout(
+				clientManagedDynamic,
+				gvrPolicy,
+				case12DepName,
+				clusterNamespace,
+				true,
+				defaultTimeoutSeconds)
+
+			return getCompliant(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+		By("Generating an compliant event on the policy")
+		managedPlc = utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12PolicyName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Normal",
+			"policy: managed/case12-config-policy",
+			"Compliant; No violation detected")
```

Comment:
Nit: this would be another candidate (in my mind) for a helper function to reduce repetition.